src/resqml2/AbstractGridRepresentation.cpp

```diff
@@ -74,15 +76,14 @@ vector<Relationship> AbstractGridRepresentation::getAllEpcRelationships() const
 	}
 
 	//Fluid
-	if (hasCellFluidPhaseUnitIndices()) {
-		std::cout << "youpieeeeee" << std::endl;
-		RESQML2_0_1_NS::RockFluidOrganizationInterpretation* fluidOrg = getFluidOrganizationInterpretation();
-		Relationship relFluid(fluidOrg->getPartNameInEpcDocument(), "", fluidOrg->getUuid());
-		relFluid.setDestinationObjectType();
-		result.push_back(relFluid);
+	if(hasCellFluidPhaseUnitIndices()) {
+		if (interp->CellFluidPhaseUnits->FluidOrganization != nullptr) {
```

Comment:
This check is not needed. The XML schema (GSOAP implementation) validates that before FEASPI objects creation.
Looking at annotation of  gsoap proxy interp->CellFluidPhaseUnits->FluidOrganization attribute tells you it is mandatory (and XML schema and UML)

Reply:
I do not understand this part. I have to use interp to do `			Relationship rel(misc::getPartNameFromReference(interp->CellFluidPhaseUnits->FluidOrganization), "", interp->CellFluidPhaseUnits->FluidOrganization->UUID);
`
But in the previous review you told me that in the partial case this will seg fault.

Reply:
For this part is the code below correct:
`	if(hasCellFluidPhaseUnitIndices()) {
		Relationship rel(misc::getPartNameFromReference(interp->CellFluidPhaseUnits->FluidOrganization), "", interp->CellFluidPhaseUnits->FluidOrganization->UUID);
		rel.setDestinationObjectType();
		result.push_back(rel);
	}`

Reply:
The check is not needed (the if). That's why I thought "interp" could go away.
But, "interp" is actually needed for the line below. It should be named "rep" or "gridRep", my mistake (again).

Reply:
Yes the proposed code looks good to me.